src/store/addressbooks.js

```diff
@@ -70,6 +70,72 @@ export function mapDavShareeToSharee(sharee) {
 	}
 }
 
+/**
+ * Sorts addressbooks by rules:
+ *  1. First group: Default personal addressbook ("contacts")
+ *  2. Second group: Recently used (based on lastUsedAddressBooks) go second, ordered from newest to oldest
+ *  3. Third group: Writeable enabled contacts, sorted by the amount of contacts they have, from more to less
+ *  4. Fourth group: Read-only enabled contacts, sorted by the amount of contacts
+ *  5. Fifth group: Disabled contacts, sorted by the amount of contacts
+ *
+ * @param {Array} addressbooks
+ * @return {Array}
+ */
+export function sortAddressbooks(addressbooks) {
+	const lastUsed = lastUsedAddressBooks(addressbooks)
+
+	return addressbooks
+		.slice()
+		.sort((a, b) => {
+			const getContactCount = (ab) => Object.keys(ab.contacts || {}).length
+
+			const getPriorityGroup = (ab) => {
+				if (ab.id === 'contacts') return 0
+				if (ab.enabled === false) return 4
+				if (lastUsed.includes(ab.id)) return 1
+				if (ab.readOnly) return 3
+				return 2
+			}
```

Comment:
coding style:

1. move this helper function out of `sortAddressbooks` so it's not redefined for every comparison
2. don't use one-line ifs. always use a { block }

Reply:
Oke